src/rules/terComputedPropertySpacingRule.ts

```diff
@@ -0,0 +1,146 @@
+    `,
+    optionsDescription: Lint.Utils.dedent`
+      The rule takes in one option, which defines to require or forbid whitespace.
+
+      * \`"never"\` (default) disallows spaces inside computed property brackets
+      * \`"always"\` requires one or more spaces inside computed property brackets
+    `,
+    options: {
+      type: 'array',
+      items: [{
+        enum: [ 'always', 'never' ]
+      }],
+      maxLength: 1
+    },
+    optionExamples: [
+      Lint.Utils.dedent`
+        "${RULE_NAME}": [true]
+        `,
+      Lint.Utils.dedent`
+        "${RULE_NAME}": [true, "always"]
+      `,
+      Lint.Utils.dedent`
+        "${RULE_NAME}": [true, "never"]
+      `
+    ],
+    typescriptOnly: false,
+    type: 'style'
+  };
+
+  private formatOptions([ alwaysOrNever ]: string[]): ITerComputedPropertySpacingOptions {
+    // handle the ruleArguments
+    return {
+      always: alwaysOrNever === 'always'
+    };
+  }
+
+  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+    const opt = this.formatOptions(this.ruleArguments);
+    const walker = new RuleWalker(sourceFile, this.ruleName, opt);
+    return this.applyWithWalker(walker);
+  }
+}
+
+class RuleWalker extends Lint.AbstractWalker<ITerComputedPropertySpacingOptions> {
+  public walk(sourceFile: ts.SourceFile) {
+    const cb = (node: ts.Node): void => {
+      if (node.kind === ts.SyntaxKind.ElementAccessExpression) {
+        this.checkElementAccessExpression(node as ts.ElementAccessExpression);
+      } else if (node.kind === ts.SyntaxKind.ComputedPropertyName) {
+        this.checkComputedPropertyName(node as ts.ComputedPropertyName);
+      }
+
+      ts.forEachChild(node, cb);
+    };
+
+    ts.forEachChild(sourceFile, cb);
+  }
+
+  private checkElementAccessExpression(node: ts.ElementAccessExpression): void {
+    const token = node.getChildAt(2);
+    return this.checkNode(node, token);
+  }
+
+  private checkComputedPropertyName(node: ts.ComputedPropertyName): void {
+    const token = node.getChildAt(1);
+    return this.checkNode(node, token);
+  }
+
+  private checkNode(node: ts.Node, token: ts.Node) {
+    const text = node.getText();
+
+    if (text.indexOf('\n') !== -1) {
+      // Rule does not apply if access spans multiple lines
+      return;
+    }
+
+    const match = /\[(.*)\]/.exec(node.getText());
+
+    if (!match) {
+      return;
+    }
+
+    const contentWithinBrackets = match[1];
+    const leftBracketIndex = node.getStart() + node.getText().indexOf('[');
+    const beforeWhitespaceLength = this.getBeforeWhitespaceLength(contentWithinBrackets);
+    const afterWhitespaceLength = this.getAfterWhitespaceLength(contentWithinBrackets);
+
+    if (this.options.always) {
+      if (beforeWhitespaceLength === 0) {
+        this.addFailureAt(leftBracketIndex, 1, this.alwaysBeforeMessage(token));
```

Comment:
This rule should be easy to add the fixer. You can look at https://github.com/buzinas/tslint-eslint-rules/blob/master/src/rules/terArrowSpacingRule.ts as an example of adding and removing the space to fix it. Once you add it, make sure you add the output property in your tests to make sure your fixer works.